name.abuchen.portfolio.ui.tests/src/name/abuchen/portfolio/ui/views/dashboard/DividendListWidgetTest.java

```diff
@@ -0,0 +1,268 @@
+package name.abuchen.portfolio.ui.views.dashboard;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertEquals;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import name.abuchen.portfolio.model.Security;
+import name.abuchen.portfolio.model.SecurityEvent;
+import name.abuchen.portfolio.model.SecurityEvent.DividendEvent;
+import name.abuchen.portfolio.money.Money;
+import name.abuchen.portfolio.ui.Messages;
+import name.abuchen.portfolio.ui.views.dashboard.DividendListWidget.DateEndRange;
+import name.abuchen.portfolio.ui.views.dashboard.DividendListWidget.DateStartRange;
+import name.abuchen.portfolio.ui.views.dashboard.DividendListWidget.DateType;
+import name.abuchen.portfolio.ui.views.dashboard.DividendListWidget.DividendItem;
+
+public class DividendListWidgetTest
```

Comment:
> My goal is always to have no warnings about not translated labels

I've set up Eclipe the way it's documented "here" and I get these no-NLS-warnings in non-test projects but - and I've just checked it right now to make sure - in test projects (here portfolio.ui.tests) these warnings are missing.

I've assumed that this is intentional because it's a test case where leniency can often be seen (e.g. I declare "throws Exception" most of the time while I hate to see this to happen in production code).

> If you annotate the class with @SuppressWarnings("nls")

I know that annotation ;-) but haven't seen any reason to use it here. I can add it, though. If I do I just have to ignore the then-showing warning that I'm using a non-required annotation ;-)